src/modules/facility-covenant/facility-covenant.controller.ts

```diff
@@ -1,17 +1,78 @@
-import { Controller, Get, Param } from '@nestjs/common';
-import { ApiBadRequestResponse, ApiInternalServerErrorResponse, ApiOkResponse, ApiOperation } from '@nestjs/swagger';
+import { Body, Controller, Get, Param, ParseArrayPipe, Post } from '@nestjs/common';
+import {
+  ApiBadRequestResponse,
+  ApiBody,
+  ApiCreatedResponse,
+  ApiInternalServerErrorResponse,
+  ApiNotFoundResponse,
+  ApiOkResponse,
+  ApiOperation,
+  ApiParam,
+} from '@nestjs/swagger';
+import { EXAMPLES } from '@ukef/constants';
 
+import { FacilityService } from '../facility/facility.service';
+import { CreateFacilityCovenantRequestDto, CreateFacilityCovenantRequestItem } from './dto/create-facility-covenant-request.dto';
+import { CreateFacilityCovenantResponseDto } from './dto/create-facility-covenant-response.dto';
 import { GetFacilityCovenantsParamsDto } from './dto/get-facility-covenants-params.dto';
 import { GetFacilityCovenantsResponseDto } from './dto/get-facility-covenants-response.dto';
 import { FacilityCovenantService } from './facility-covenant.service';
 
 @Controller()
 export class FacilityCovenantController {
-  constructor(private readonly facilityCovenantService: FacilityCovenantService) {}
+  constructor(private readonly facilityCovenantService: FacilityCovenantService, private readonly facilityService: FacilityService) {}
+
+  @Post('facilities/:facilityIdentifier/covenants')
+  @ApiOperation({
+    summary: 'Create a new covenant for a facility.',
+  })
+  @ApiParam({
+    name: 'facilityIdentifier',
+    required: true,
+    type: 'string',
+    description: 'The identifier of the facility in ACBS.',
+    example: EXAMPLES.FACILITY_ID,
+  })
+  @ApiBody({
+    type: CreateFacilityCovenantRequestItem,
+    isArray: true,
+  })
+  @ApiCreatedResponse({
+    description: 'The covenant has been successfully created.',
+    type: CreateFacilityCovenantResponseDto,
+  })
+  @ApiNotFoundResponse({
+    description: 'The facility was not found.',
+  })
+  @ApiBadRequestResponse({
+    description: 'Bad request.',
+  })
+  @ApiInternalServerErrorResponse({
+    description: 'An internal server error has occurred.',
+  })
+  async createCovenantForFacility(
+    @Param() params: GetFacilityCovenantsParamsDto,
```

Comment:
If we reuse GET params DTO, maybe we should rename it to make more generic, remove Get prefix.

Reply:
Good spot, changed.